src/assign.cpp

```diff
@@ -53,34 +53,56 @@ fba::Assignment::Assignment(fba::Targets::pshr tgs,
     for (auto const & tp : tgtypes) {
         nassign_tile[tp].clear();
         nassign_petal[tp].clear();
+        nassign_slitblock[tp].clear();
         for (size_t t = 0; t < ntile; ++t) {
             int32_t tile_id = tiles_->id[t];
             nassign_tile[tp][tile_id] = 0;
             nassign_petal[tp][tile_id].clear();
             for (int32_t p = 0; p < hw_->npetal; ++p) {
                 nassign_petal[tp][tile_id][p] = 0;
             }
+            for (int32_t p = 0; p < hw_->npetal; ++p) {
+                nassign_slitblock[tp][tile_id][p].clear();
+                for (int32_t s = 0; s < hw_->nslitblock; ++s) {
+                    nassign_slitblock[tp][tile_id][p][s] = 0;
+                }
+            }
             tile_target_xy[tile_id].clear();
```

Comment:
This change looks messier than it should -- I dropped a level of indentation by changing the "if (tp == TARGET_TYPE_SKY) { ... }" to a check-continue "if (tp != TARGET_TYPE_SKY) continue; ...".

The check to not count ETC fibers is new -- here, I'm doing that by checking their slitblock value, which happens to be -1; perhaps it would be more clear to check explicitly the `hw_->loc_device_type[loc]`.

(In addition, in the `stucksky.py` code, we now only check POS fibers anyway, so ETC fibers should not exist in the stuck_sky[tile_id] mapping; so the ETC check is just defensive coding.)

Really the only per-slitblock change is incrementing the `nassign_slitblock` at the bottom!

Reply:
Thanks for the helpful explanation ;-)